src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java

```diff
@@ -629,6 +629,21 @@ public ResponseEntity<Course> getCourseWithOrganizations(@PathVariable Long cour
         return ResponseEntity.ok(course);
     }
 
+    /**
+     * GET /courses/:courseId/with-learning-paths Get a course by id with eagerly loaded learning paths
+     *
+     * @param courseId the id of the course
+     * @return the course with eagerly loaded learning paths
+     */
+    @GetMapping("courses/{courseId}/with-learning-paths")
+    @EnforceAtLeastInstructor
+    public ResponseEntity<Course> getCourseWithLearningPaths(@PathVariable Long courseId) {
+        log.debug("REST request to get a course with its organizations : {}", courseId);
+        Course course = courseRepository.findWithEagerLearningPathsByIdElseThrow(courseId);
+        authCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, course, null);
+        return ResponseEntity.ok(course);
```

Comment:
Maybe it is enough here to return a list of LearningPathDTOs and not the full course?